Structural/Flyweight/FlyweightInterface.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace DesignPatterns\Structural\Flyweight;
+
+/**
+ * An interface through which flyweights can receive and act on extrinsic state
+ */
+interface FlyweightInterface {
```

Comment:
I guess opening brace in class/interface/trait declaration should go on new line.